matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/profile/DefaultProfileService.kt

```diff
@@ -63,6 +81,41 @@ internal class DefaultProfileService @Inject constructor(private val taskExecuto
                 .executeBy(taskExecutor)
     }
 
+    override fun updateAvatar(userId: String, newAvatarUri: Uri, fileName: String, matrixCallback: MatrixCallback<Unit>): Cancelable {
+        val cancelableBag = CancelableBag()
+        val workerParams = UploadAvatarWorker.Params(sessionId, newAvatarUri, fileName)
+        val workerData = WorkerParamsFactory.toData(workerParams)
+
+        val uploadAvatarWork = workManagerProvider.matrixOneTimeWorkRequestBuilder<UploadAvatarWorker>()
```

Comment:
I don't think we need to make uploading avatar uses a worker. It's alright if it fails, we'll get an error and can retry later. What do you think about that @bmarty ?

Reply:
I think this code is more complicated that I'd expect it to be. We should use the uploadService facilities. I have not checked in the code if it is possible.